### PR TITLE
[Improvement] * - IPv6 support for subnets, public load balancers, and CloudFront (breaking change)

### PR DESCRIPTION
**Pull request by michaelwittig**
predecessor https://github.com/widdix/aws-cf-templates/pull/182

### CODE REVIEW

File: docs/vpc.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
...
+A VPC is a virtual network inside AWS where you can isolate your workload. A VPC consists of several subnets. Each subnet is bound to an Availability Zone. A **public** subnet has a direct route to/from the Internet. As long as your EC2 instances have an public IPv4 address (default) or IPv6 address, they can communicate (in and out) with the Internet. A **private** subnet does not have a IPv4 route to/from the Internet but an Ipv6 route to the Internet exists. Instances in private subnets can not be accessed from the public Internet. If you want to access the Internet from a private subnetyou need to create a NAT gateway/instance or assign an IPv6 address. You can deploy a bastion host/instance to reduce the attack surface of internal applications.
```
Comment (andreaswittig) [inline docs/vpc.md:5]: `subnetyou` => `subnet you`

File: ecs/service-cluster-alb.yaml [hunk 2]
```diff
@@ -165,6 +165,19 @@ Resources:
       - SubDomainNameWithDot: !Ref SubDomainNameWithDot
         HostedZoneName: {'Fn::ImportValue': !Sub '${ParentZoneStack}-HostedZoneName'}
       Type: A
+  RecordSetIPv6: # We can not conditionally create this only if the cluster's ALB has IPv6 turned on. Route53 does not let us query a broken AAAA record either. It just shows up as a Route53 record.
```
Comment (andreaswittig) [inline ecs/service-cluster-alb.yaml:168]: `It just shows up as a Route53 record.` What does that mean?

File: vpc/vpc-2azs-legacy.yaml [hunk 3]
```diff
@@ -42,6 +43,10 @@ Parameters:
   CidrBlock:
     Description: 'The set of IP addresses for the VPC (e.g., 10.0.0.0/16).'
     Type: String
+  CidrBlockIPv6:
+    Description: 'The set of IPv6 addresses for the VPC (e.g., 2600:1f18:277f:e700::/56; set to 2001:db8::/32 if yor VPC does not support IPv6).'
```
Comment (andreaswittig) [inline vpc/vpc-2azs-legacy.yaml:47]: `yor` => `your`

File: ecs/service-cluster-alb.yaml (hunk 2 above)
Comment (michaelwittig) [reply ecs/service-cluster-alb.yaml:168]: you see it in Route53 as a record but you can not resolve it.

**Pull request merged**